src/components/FastImage/index.js

```diff
@@ -0,0 +1,46 @@
+import React, {useEffect, useMemo} from 'react';
+import {Image} from 'react-native';
+import addEncryptedAuthTokenToURL from '../../libs/addEncryptedAuthTokenToURL';
+
+const RESIZE_MODES = {
+    contain: 'contain',
+    cover: 'cover',
+    stretch: 'stretch',
+    center: 'center',
+};
+
+const FastImage = (props) => {
+    // eslint-disable-next-line
+    const {source, onLoad, ...rest} = props;
+
+    // Check for headers - if it has them then we need to instead just add the
+    // encryptedAuthToken to the url as RNW's Image component headers do not properly cache
+    const imageSource = useMemo(() => {
+        if (typeof source === 'number' || source.headers == null) {
+            return source;
+        }
+        return {uri: addEncryptedAuthTokenToURL(source.uri)};
+    }, [source]);
+
+    // Conform DOM onLoad event to return width and height to match RNFastImage
+    // https://github.com/DylanVann/react-native-fast-image#onload-event--void
+    useEffect(() => {
+        if (onLoad == null) {
+            return;
+        }
+        const uri = typeof imageSource === 'number'
+            ? Image.resolveAssetSource(imageSource).uri
+            : imageSource.uri;
+        Image.getSize(uri, (width, height) => {
+            onLoad({nativeEvent: {width, height}});
+        });
+    }, [imageSource, onLoad]);
```

Comment:
It looks like we don't "need" to use hooks here, right? We try not to use hooks in our app unless absolutely necessary (a.k.a. if a 3rd party lib has only has some functionality in the form of a hook) 😬

Reply:
Yeah maybe @kidroca might have a better idea here - I originally was parsing the `naturalWidth` and `naturalHeight` from the `Image` `onLoad` prop, but couldn't see a nice way of doing this between Chrome and Safari so decided it would be better to have a side-effect that still just used `Image.getSize` whenever the image source changed.

We could run its contents on the `onLoad` prop, but then it would have to wait for the image component to finish loading first.

Reply:
Sorry for the slow response 😅 I actually think it's totally fine doing what you're doing here, but I'm just being super picky about doing it in a functional component w/ `useEffect` vs a class based component - since in our style guide we say we should try to never use hooks :D

Reply:
Ace, I'll convert this to a class component then 🙌

Reply:
Awesome, thanks so much!

Reply:
FYI I'm referencing style guidelines from [here](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#are-we-allowed-to-use-insert-brand-new-react-feature-why-or-why-not):

> Hooks - Use a class `Component` and relevant lifecycle methods instead of hooks. One exception here is if a 3rd party library offers some functionality via hooks (and only hooks).